src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners.Implementation
+{
+    using System;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.Extensions.DiagnosticAdapter;
+
+    /// <summary>
+    /// <see cref="IApplicationInsightDiagnosticListener"/> implementation that listens for evens specific to EntiryFrameworkCore
+    /// </summary>
+    public class EntityFrameworkDiagnosticListener : IApplicationInsightDiagnosticListener
+    {
+        private readonly TelemetryClient client;
+        private readonly ContextData<long> beginDependencyTimestamp = new ContextData<long>();
+
+        public string ListenerName { get; } = "Microsoft.EntityFrameworkCore";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:HostingDiagnosticListener"/> class.
+        /// </summary>
+        /// <param name="client"><see cref="TelemetryClient"/> to post traces to.</param>
+        public EntityFrameworkDiagnosticListener(TelemetryClient client)
+        {
+            this.client = client;
+        }
+
+        [DiagnosticName("Microsoft.EntityFrameworkCore.BeforeExecuteCommand")]
+        public void OnBeginCommand(long timestamp)
+        {
+            beginDependencyTimestamp.Value = timestamp;
+        }
+
+        [DiagnosticName("Microsoft.EntityFrameworkCore.AfterExecuteCommand")]
+        public void OnEndCommand(IDbCommand command, string executeMethod, Guid instanceId, long timestamp, bool isAsync)
+        {
+            var start = beginDependencyTimestamp.Value;
+            var end = timestamp;
+
+            var telemetry = new DependencyTelemetry();
+            telemetry.Name = command.Connection.Database;
+            telemetry.Data = command.CommandText;
+            telemetry.Duration = new TimeSpan(end - start);
+            telemetry.Timestamp = DateTimeOffset.Now - telemetry.Duration;
```

Comment:
We had a discussion the other day that we should probably go through our code and replace all instances of DateTimeOffset.Now with DateTimeOffset.UtcNow because there happens to be a perf difference between them so maybe this one should be updated now.